clientconn.go

```diff
@@ -1141,10 +1141,15 @@ func (cc *ClientConn) Close() error {
 
 	<-cc.resolverWrapper.serializer.Done()
 	<-cc.balancerWrapper.serializer.Done()
-
+	var wg sync.WaitGroup
 	for ac := range conns {
-		ac.tearDown(ErrClientConnClosing)
+		wg.Add(1)
```

Comment:
Is there any motivation behind replacing the tear down process with goroutines with `sync.WaitGroup`? Is this tearDown going to be heavy ops, if not i feel earlier approach might be better? WDYT?

Reply:
This change was suggested by Doug, because now we wait for all go routines inside transport to exit, so it will take time if we serially wait for all transports to close. So we changed it to do the following:
-we will instruct all transports to close simultaneously
- All the transports will wait for all go routines inside it to exit
- we wait for all transports to close before exiting 
Instead for asking each transport to close serially , as other transport might not even be signalled to close umtil all resources in previous transport are cleaned up. Since the transports are not dependetn on each other, this approach saves time.